src/frequenz/sdk/util/ringbuffer.py

```diff
@@ -0,0 +1,259 @@
+# License: MIT
+# Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+"""Ringbuffer implementation with focus on time & memory efficiency."""
+
+from __future__ import annotations
+
+from copy import deepcopy
+from datetime import datetime, timedelta
+from typing import Any, Generic, Sequence, TypeVar, Union
+
+import numpy as np
+
+T = TypeVar("T")
+
+Container = Union[list, np.ndarray]
+
+
+class OrderedRingBuffer(Generic[T]):
+    """Time aware ringbuffer that keeps its entries sorted by time."""
+
+    def __init__(
+        self,
+        buffer: Any,
```

Comment:
Well, it can't be `int` as it should be a container (as the docu says), more specifically list or nd array.

I tried using a more explicit type hint (e.g. `:list | np.ndarray`), but it proves rather difficult to do if you want to support multiple storage containers.

At the end i didn't find a solution that would work so far...

Reply:
Did you try `typing.Union`?

Reply:
It is working now, not sure why it didn't earlier. Will updated when the other changes are done